### ISSUE TITLE
Mocks aren't cleared when using Junit 5's @Nested classes

### ISSUE THREAD

**Issue by checketts** [Ninja-Squad/springmockk#12]
I'm super excited about leveraging `@MockkBean`. Thanks for creating it!

I ran into a bug where my mocks weren't being cleared when using `@Nested`

I was able to work around it by avoiding `@Nested`classes for testing.  It appears that `@MockBean` had a similar problem: https://github.com/spring-projects/spring-boot/issues/12470

**Comment by jnizet** [comment]
Thanks for the kind words. It's indeed a Spring Boot problem.
An easy workaround with MockK is tosimply reset all mocks after each test. If you don't rely on global mocks that must not be reset, this works fine. Here's what I use to do that:

```
import io.mockk.clearAllMocks
import org.junit.jupiter.api.extension.AfterTestExecutionCallback
import org.junit.jupiter.api.extension.ExtensionContext

/**
 * Junit extension that resets all mocks after every test execution.
 * It should not be needed because MockkBean clears the mocks that have been created in the application context by
 * default. Unfortunately, that doesn't work for nested tests due to a Spring Boot issue
 * (see https://github.com/spring-projects/spring-boot/issues/12470). So, to make things easier, we keep using this
 * extension.
 * @author JB Nizet
 */
class ClearAllMocksExtension : AfterTestExecutionCallback {
    override fun afterTestExecution(context: ExtensionContext?) {
        clearAllMocks()
    }
}
```

**Comment by checketts** [comment]
Any reason not to submit this extension to mockk or to include in the springmockk?

**Comment by jnizet** [comment]
This extension is specific to JUnit 5, and is quite easy to create, so I'm not sure MockK would create a JUnit5 module or additional dependency just to host this extension.

Ading it to SpringMockK has the same problem: I would have to add a module, or an optional dependency, just to contain this workaround for a Spring Boot bug (that doesn't work in all cases). Hopefully this bug will be fixed soon, and the workaround won't be needed anymore.

**Comment by jnizet** [comment]
Fixed in Spring Boot 2.4.0 / SpringMockK 3.0.0
